tools/ci_build/build.py

```diff
@@ -1547,6 +1547,12 @@ def build_protoc_for_host(cmake_path, source_dir, build_dir, args):
     elif is_macOS():
         if args.use_xcode:
             cmd_args += ['-G', 'Xcode']
+            # CMake < 3.18 has a bug setting system arch to arm64 (if not specified) for Xcode 12,
+            # protoc for host should be built using host architecture
+            # Explicitly specify the CMAKE_OSX_ARCHITECTURES for x86_64 Mac.
```

Comment:
Only set DCMAKE_OSX_ARCHITECTURES for x86_64 Mac,
Hopefully the new Mac using Apple Silicon will use CMake 3.18+ which does not have this issue, or a new ORT release.